sdks/python/apache_beam/runners/interactive/caching/streaming_cache.py

```diff
@@ -247,8 +247,11 @@ def __init__(
     self._sample_resolution_sec = sample_resolution_sec
     self._is_cache_complete = is_cache_complete
 
+    from apache_beam.runners.interactive import interactive_beam as ib
     if cache_dir:
       self._cache_dir = cache_dir
+    elif ib.options.specified_cache_dir:
+      self._cache_dir = ib.options.specified_cache_dir
```

Comment:
I like to call code like this "spooky action at a distance". In general, configuration should be passed in through the constructor. This makes it obvious as to what settings the object needs. Accessing a global variable within the constructor esp. with an existing cache_dir argument can make it confusing for maintainers. I recommend moving this logic to where the StreamingCache and CacheManager are created and the cache_dir parameter is passed.